File: src/test/csharp/eu/sig/training/ch10/PerfectPictureMockTest.cs
using NUnit.Framework;
using System.Drawing;
using eu.sig.training.ch06.simpledigitalcamera;
using Moq;

namespace eu.sig.training.ch10
{
    [TestFixture]
    public class PerfectPictureMoqTest
    {
        // tag::testNightPictureMockito[]
        [Test]
        public void TestNightPictureMoq()
        {
            Image image =
                Image.FromFile("../../../../test/resources/VanGoghStarryNight.jpg");
            var cameraMock = new Mock<ISimpleDigitalCamera>();
            cameraMock.Setup(foo => foo.TakeSnapshot()).Returns(image);
            PerfectPicture.camera = cameraMock.Object;
            Assert.AreSame(image, new PerfectPicture().TakePerfectPicture(0));
            cameraMock.Verify(foo => foo.FlashLightOn(), Times.AtMostOnce());
        }
        // end::testNightPictureMockito[]
    }
}

File: src/test/java/eu/sig/training/ch10/PerfectPictureMockTest.java
package eu.sig.training.ch10;

import static org.junit.Assert.assertEquals;
import static org.mockito.Mockito.mock;
import static org.mockito.Mockito.verify;
import static org.mockito.Mockito.when;

import java.awt.image.BufferedImage;
import java.io.File;
import java.io.IOException;

import javax.imageio.ImageIO;

import org.junit.Test;

import eu.sig.training.ch06.simpledigitalcamera.SimpleDigitalCamera;
import eu.sig.training.ch10.PerfectPicture;

public class PerfectPictureMockTest {
    // tag::testNightPictureMockito[]
    @Test
    public void testNightPictureMockito() throws IOException {
        BufferedImage image =
            ImageIO.read(new File("src/test/resources/VanGoghStarryNight.jpg"));
        SimpleDigitalCamera cameraMock = mock(SimpleDigitalCamera.class);
        PerfectPicture.camera = cameraMock;
        when(cameraMock.takeSnapshot()).thenReturn(image);
        assertEquals(image, new PerfectPicture().takePerfectPicture(0));
        verify(cameraMock).flashLightOn();
    }
    // end::testNightPictureMockito[]
}
